plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java

```diff
@@ -174,4 +166,30 @@ public void refresh() {
             SocketAccess.doPrivilegedVoid(credentials::refresh);
         }
     }
+
+    private static class InternalAmazonS3Wrapper extends AbstractRefCounted implements AmazonS3Wrapper {
```

Comment:
I don't think we need the `AmazonS3Wrapper` interface just make this class public and name it `AmazonS3Wrapper`

Reply:
I'd call it `AmazonS3Reference`